src/lib/services/searchUtilActions/AasSearcher.ts

```diff
@@ -0,0 +1,221 @@
+    registryShellDescriptorEntries?: AssetAdministrationShellDescriptor[] | null;
+    shellsByRegistryEndpoint?: { path: string; aas: AssetAdministrationShell }[] | null;
+    shellsSavedInTheRepositories?: INullableAasRepositoryEntries[] | null;
+    submodelsSavedInTheRepository?: Submodel[] | null;
+    entitiesInMultipleDataSources?: NullableMultipleDataSourceSetupParameters | null;
+    log?: Log | null;
+}
+
+export type AasData = {
+    submodelDescriptors: SubmodelDescriptor[] | undefined;
+    aasRegistryRepositoryOrigin: string | undefined;
+};
+export type AasSearchResult = {
+    redirectUrl: string;
+    aas: AssetAdministrationShell | null;
+    aasData: AasData | null;
+};
+
+export class AasSearcher {
+    private constructor(
+        protected readonly discoveryServiceClient: IDiscoveryServiceApi,
+        protected readonly registryService: IRegistryServiceApi,
+        protected readonly multipleDataSource: MultipleDataSource,
+        protected readonly fetch: (input: RequestInfo | URL, init?: RequestInit | undefined) => Promise<Response>,
+        protected readonly log: Log,
+    ) {}
+
+    static create(_baseUrl: string = ''): AasSearcher {
+        const multipleDataSource = MultipleDataSource.create();
+        const registryServiceClient = RegistryServiceApi.create(process.env.REGISTRY_API_URL, mnestixFetch());
+        const discoveryServiceClient = DiscoveryServiceApi.create(process.env.DISCOVERY_API_URL, mnestixFetch());
+        const log = Log.create();
+        return new AasSearcher(discoveryServiceClient, registryServiceClient, multipleDataSource, fetch, log);
+    }
+
+    static createNull({
+        discoveryEntries = [],
+        registryShellDescriptorEntries = [],
+        shellsByRegistryEndpoint = [],
+        shellsSavedInTheRepositories = [],
+        submodelsSavedInTheRepository = [],
+        log = null,
+    }: NullableSearchSetupParameters = {}): AasSearcher {
+        const stubbedFetch = async (input: RequestInfo | URL): Promise<Response> => {
+            if (!shellsByRegistryEndpoint) return Promise.reject(new Error('no registry configuration'));
+            for (const aasEntry of shellsByRegistryEndpoint) {
+                if (aasEntry.path === input) return new Response(JSON.stringify(aasEntry.aas));
+            }
+            return Promise.reject(new Error('no aas for on href:' + input));
+        };
+        return new AasSearcher(
+            DiscoveryServiceApi.createNull({ discoveryEntries: discoveryEntries }),
+            RegistryServiceApi.createNull({ registryShellDescriptorEntries }),
+            MultipleDataSource.createNull({
+                shellsByRegistryEndpoint,
+                shellsSavedInTheRepositories: shellsSavedInTheRepositories,
+                submodelsSavedInTheRepository,
+            }),
+            stubbedFetch,
+            log ?? Log.createNull(),
+        );
+    }
+
+    async fullSearch(val: string): Promise<AasSearchResult> {
```

Comment:
I'd also include that we are searching an AAS here in the naming. Maybe "fullSearchForAas" or something like that